src/new/wallet.js

```diff
@@ -343,38 +409,40 @@ class HathorWallet extends EventEmitter {
    * It is also called if the network is down.
    *
    * @param {Number} newState Enum of new state after change
-   **/
+   */
   async onConnectionChangedState(newState) {
     if (newState === ConnectionState.CONNECTED) {
       this.setState(HathorWallet.SYNCING);
 
       try {
-      // If it's the first connection we just load the history
-      // otherwise we are reloading data, so we must execute some cleans
-      // before loading the full data again
-      if (this.firstConnection) {
-        this.firstConnection = false;
-        const walletData = await this.storage.getWalletData();
-        await syncHistory(0, walletData.gapLimit, this.storage, this.conn);
-      } else {
-        if (this.beforeReloadCallback) {
-          this.beforeReloadCallback();
+        // If it's the first connection we just load the history
+        // otherwise we are reloading data, so we must execute some cleans
+        // before loading the full data again
+        if (this.firstConnection) {
+          this.firstConnection = false;
+          const addressesToLoad = await scanPolicyStartAddresses(this.storage);
```

Comment:
This is agnostic to the configured scanning policy, so to add more scanning policies we should configure them on the `scanPolicyStartAddresses`